publish fails with subpackages

The package [lpd](https://github.com/Chris00/lpd) declares a subpackage [lpd.socket](https://github.com/Chris00/lpd/blob/master/src/jbuild#L12) and this makes `dune-release publish` fail with
```
        odoc _doc/_odoc/pkg/lpd/page-index.odoc
File "../../../_mlds/lpd/index.mld", line 3, characters 0-22:
only one title-level heading is allowed
File "../../../_mlds/lpd/index.mld", line 3, characters 0-22:
'1': bad section level (2-4 allowed)
dune-release: [ERROR] No package delegate found. Try `dune-release help
                      delegate` for more information.
```
because `index.mld` content is
```
{1 Library lpd}
The entry point of this library is the module:
{!module-Lpd}.
{1 Library lpd.socket}
The entry point of this library is the module:
{!module-Socket}.
```

--- a/lib/pkg.ml
+++ b/lib/pkg.ml
@@ -60,6 +60,9 @@ let opam_doc p = opam_field_hd p "doc"
 let opam_homepage_sld p = opam_homepage p >>| function
   | None -> None
   | Some uri -> match uri_sld uri with None -> None | Some sld -> Some (uri, sld)
+let opam_doc_sld p = opam_doc p >>| function
+  | None -> None
+  | Some uri -> match uri_sld uri with None -> None | Some sld -> Some (uri, sld)
 
 let name p = Ok p.name
 
@@ -68,27 +71,45 @@ let version p = match p.version with
 | None -> Vcs.get () >>= fun r -> Vcs.describe ~dirty:false r
 
 let delegate p =
-  let not_found () =
-    R.error_msg "No package delegate found. \
-                 Try `dune-release help delegate` for more information."
+  let not_found = function
+  | None ->
+    R.error_msg
+      "Package delegate command cannot be found (no homepage or doc field). \
+       Try `dune-release help delegate` for more information."
+  | Some cmd ->
+    R.error_msgf
+      "%a: package delegate cannot be found. \
+       Try `dune-release help delegate` for more information."
+      Cmd.pp cmd
   in
   match p.delegate with
   | Some cmd -> Ok (Some cmd)
   | None ->
-      match OS.Env.(value "DUNE_RELEASE_DELEGATE" (some cmd) ~absent:None) with
-      | Some cmd -> Ok (Some cmd)
-      | None ->
-          opam_homepage_sld p >>= function
-          | None -> not_found ()
-          | Some (_, sld) ->
-              let exec = strf "%s-dune-release-delegate" sld in
-              let cmd = Cmd.v exec in
-              OS.Cmd.exists cmd >>= function
-              | true -> Ok (Some cmd)
-              | false ->
-                  if exec <> "github-dune-release-delegate"
-                  then not_found ()
-                  else Ok None
+      let delegate =
+        match OS.Env.(value "DUNE_RELEASE_DELEGATE" (some string) ~absent:None) with
+        | Some cmd -> Some cmd
+        | None     -> None
+      in
+      let guess_delegate () =
+        match delegate with
+        | Some d -> Ok d
+        | None   ->
+            let cmd sld = strf "%s-dune-release-delegate" sld in
+            (* first look at `doc:` then `homepage:` *)
+            opam_doc_sld p >>= function
+            | Some (_, sld) -> Ok (cmd sld)
+            | None -> opam_homepage_sld p >>= function
+              | Some (_, sld) -> Ok (cmd sld)
+              | None -> not_found None
+      in
+      guess_delegate () >>= fun cmd ->
+      let x = Cmd.v cmd in
+      OS.Cmd.exists x >>= function
+      | true -> Ok (Some x)
+      | false ->
+          if cmd <> "github-dune-release-delegate"
+          then not_found (Some x)
+          else Ok None
 
 let build_dir p = match p.build_dir with
 | Some b -> Ok b

